### PR TITLE
Use pixi

### PR DESCRIPTION
**Pull request by brauliorivas**
BEGINRELEASENOTES
- #63 serves as inspiration for this PR. This PR will contain a migration to [pixi](https://pixijs.com/). The previous PR was sort of "prototype" for testing, and this one serves as the real implementation. 
- I'll start by working on the `MCParticle Tree` view because it requires the most performance improvements. Also, we have a huge file to use for testing performance. Other views will also be reworked, but I want an MVP for this view.  

ENDRELEASENOTES

### CODE REVIEW

**Comment by brauliorivas** [comment]
I've linked #44 because of pixi logic, I had to change to async drawing, so the images (converted to SVG with MathJax for MCParticle) now won't show after being loaded even if the user changes between events rapidly.

**Comment by brauliorivas** [comment]
> * When playing with few views I noticed that when I grab some box the box gets shifted such that the cursor is right in the middle of the box (both x and y coordinates). Would it be possible to not do this and dragged object won't move in respect to the cursor?

Implemented.
> * The pixel density is low for the connection line, it is especially visible for very long ones. Can it be somehow increased? Will this affect performance?

Sure, I've activated antialias which solves this problem. In theory it affects performance, but is not too noticeable. 

> * Can the overlay be somewhat postponed? I mean, so it will not pop up right away.

I've added a 500ms delay.

> * I'm not sure if this is a limitation of my browser or it is a setting in pixijs, but I tend to loose hold of the box quite often. If I drag the box quickly at some point it will stop moving and I no longer drag it.

It was because of how the event listener for `mousemove` worked. It used to be added to the box itself, but now is added to the canvas, so now when dragging, the box is not lost.

**Comment by brauliorivas** [comment]
> When I try to draw the MCParticles for this event I run out of memory and my pc freezes until the OOM killer kills my browser. Can we try to avoid that? The drawing of the event also does not work in the current version but at least nothing crashes :)
> 
> [test.edm4hep.json](https://github.com/user-attachments/files/16349336/test.edm4hep.json)

Pixi wasn't performant enough because I was rendering objects that were not visible. But now I've implemented the culling [system](https://pixijs.com/8.x/guides/basics/scene-graph#culling) correctly and can visualize that event (without crashing). It  can even visualize this event too [test_mc_shower.edm4hep.json](https://github.com/user-attachments/files/15585039/test_mc_shower.edm4hep.json), which used to make eede crash (so the last optimizations definitely helped by a lot). However there are two problems:
- Still consumes a lot of memory, so I have to find a way to reduce this. 
- When visualizing those events, pixi fails to load the images

Overall, being able to at least visualize those events makes a big difference.

File: css/main.css [hunk 1]
```diff
@@ -4,6 +4,7 @@ body {
   /* overflow: hidden; */
   font-family: sans-serif;
   font-size: 16px;
+  overflow: hidden;
```
Comment (kjvbrt) [inline css/main.css:7]: Can you uncomment the line 4 instead?

**Comment by kjvbrt** [comment]
Screenshot from 2024-07-26 15-41-29

The box and the connecting line gets "disconnected" quite often. And also the gap size changes depending of the box position.

**Comment by kjvbrt** [comment]
When I use this json file 
[out_sim_gun_10.edm4hep.json](https://github.com/user-attachments/files/16392733/out_sim_gun_10.edm4hep.json)

I gen errors in the JS console.

**Pull request merged**